### PR TITLE
Pass useBytes in updateVerticesDataDirectly in Geometry & VertexBuffer

### PR DESCRIPTION
**Pull request by a-carvallo**
Related to : https://github.com/BabylonJS/Babylon.js/pull/4026#pullrequestreview-108884133

### CODE REVIEW

File: src/Mesh/babylon.vertexBuffer.ts [hunk 1]
```diff
@@ -232,10 +232,13 @@
...
          * Updates directly the underlying WebGLBuffer according to the passed numeric array or Float32Array.  
-         * Returns the directly updated WebGLBuffer. 
+         * Returns the directly updated WebGLBuffer.
+         * @param data the new data
+         * @param offset the new offset
+         * @param useBytes set to true if the offset is in bytes
          */
-        public updateDirectly(data: DataArray, offset: number): void {
-            return this._buffer.updateDirectly(data, offset);
+        public updateDirectly(data: DataArray, offset: number, useBytes: boolean = false): void {
```
Comment (deltakosh) [inline src/Mesh/babylon.vertexBuffer.ts]: Just realized that it returns void so we should remove the returns comment and the  return keyword:)

File: src/Mesh/babylon.buffer.ts [hunk 2]
```diff
@@ -123,13 +123,13 @@
...
          */
-        public updateDirectly(data: DataArray, offset: number, vertexCount?: number, useBytes = false): void {
+        public updateDirectly(data: DataArray, offset: number, vertexCount?: number, useBytes: boolean = false): void {
             if (!this._buffer) {
                 return;
             }
 
             if (this._updatable) { // update buffer
-                this._engine.updateDynamicVertexBuffer(this._buffer, data, useBytes ? offset : offset * 4, (vertexCount ? vertexCount * this.byteStride : undefined));
+                this._engine.updateDynamicVertexBuffer(this._buffer, data, useBytes ? offset : offset * this.byteStride, (vertexCount ? vertexCount * this.byteStride : undefined));
```
Comment (bghgary) [inline src/Mesh/babylon.buffer.ts]: I don't believe this is correct. If `useBytes` is false, then it should behave like before where `offset` is in float32 units which is a multiple of 4. `byteStride` has nothing to do with it.

Comment (a-carvallo) [reply src/Mesh/babylon.buffer.ts]: Shouldn't it be the same  for `vertexCount` then ?

Comment (deltakosh) [reply src/Mesh/babylon.buffer.ts]: That's right.I missed this one

Comment (deltakosh) [reply src/Mesh/babylon.buffer.ts]: For vertex count we are good as the bytestride is the stride in byte of the entire vertex

Comment (a-carvallo) [reply src/Mesh/babylon.buffer.ts]: So 4 for both. (I guess I was assuming it may be different, but actually it all ends in an array of Float32)

Comment (deltakosh) [reply src/Mesh/babylon.buffer.ts]: Actually only one not both (vertex count remains unchanged)

Comment (a-carvallo) [reply src/Mesh/babylon.buffer.ts]: Could `Float32Array.BYTES_PER_ELEMENT` be used in that case ? Unless I am wrong it's exactly what the 4 here means. (It helps understand better I think).

Comment (deltakosh) [reply src/Mesh/babylon.buffer.ts]: Works for me

Comment (bghgary) [reply src/Mesh/babylon.buffer.ts]: Perhaps update the other 4s in the file to this constant?

**Pull request merged**